### PR TITLE
ArC: custom context can get CurrentContextFactory during instantiation

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/context/CustomContextTest.java [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+        }
+
+        @Override
+        public ContextState getState() {
+            throw new UnsupportedOperationException();
+        }
+
+    }
+
+    public class InvalidConcstructorContext implements InjectableContext {
```
Comment (Ladicek) [inline independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/context/CustomContextTest.java:270]:
```suggestion
    public class InvalidConstructorContext implements InjectableContext {
```

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/context/CustomContextTest.java [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+import io.quarkus.arc.processor.ContextConfigurator;
+import io.quarkus.arc.processor.ContextRegistrar;
+import io.quarkus.arc.test.ArcTestContainer;
+
+public class CustomContextTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = ArcTestContainer.builder()
+            .beanClasses(FieldScoped.class, MeadowScoped.class, Mina.class, InvalidNestedContext.class,
+                    InvalidAbstractContext.class, InvalidAbstractContext.class, FieldContext.class, MeadowContext.class)
```
Comment (Ladicek) [inline independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/context/CustomContextTest.java:38]:
```suggestion
                    InvalidAbstractContext.class, InvalidConstructorContext.class, FieldContext.class, MeadowContext.class)
```

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/context/CustomContextTest.java (hunk 1 above)
Comment (mkouba) [reply independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/context/CustomContextTest.java:270]: :facepalm: thanks

**Review by Ladicek — APPROVED** [review]
LGTM

File: independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ContextCreator.java [hunk 3]
```diff
@@ -7,7 +7,10 @@
  */
 public interface ContextCreator {
 
+    public static final String KEY_CURRENT_CONTEXT_FACTORY = "io.quarkus.arc.currentContextFactory";
```
Comment (manovotn) [inline independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ContextCreator.java:10]: `public static final` is redundant when declared on an interface.

**Review by manovotn — APPROVED** [review]
Left just a nitpicker note, otherwise LGTM

File: independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ContextCreator.java (hunk 3 above)
Comment (mkouba) [reply independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ContextCreator.java:10]: Ah, of course...

**Pull request merged**